dali/pipeline/data/buffer.h

```diff
@@ -302,19 +314,19 @@ class Buffer {
 
   Backend backend_;
 
-  TypeInfo type_;  // Data type of underlying storage
-  shared_ptr<void> data_;  // Pointer to underlying storage
-  Index size_;  // The number of elements in the buffer
-  bool shares_data_;
+  TypeInfo type_ = {};  // Data type of underlying storage
+  shared_ptr<void> data_ = nullptr;  // Pointer to underlying storage
+  Index size_ = 0;  // The number of elements in the buffer
+  bool shares_data_ = false;
 
   // To keep track of the true size
   // of the underlying allocation
-  size_t num_bytes_;
+  size_t num_bytes_ = 0;
 
-  bool pinned_;  // Whether the allocation uses pinned memory
+  bool pinned_ = true;  // Whether the allocation uses pinned memory
```

Comment:
Move it next to shares_data, save 8 bytes in structure size.

Reply:
The best layout I can think of is:
```
type_ (whatever, aligned to 8 bytes)
data_ (0... 16)
size_ (16..24)
num_bytes (24..32)
device_(32..36)
pinned_(36..37)
shares_data_(37..38)
(pad: 38..40)
```
Now we have
```
data_ (0..16)
size_ (16..24)
shares_data_ (24..25)
(pad: 25..32)
num_bytes_ (32..40)
pinned_ (40..41)
(pad: 41..44)
device (44..48)
```

Reply:
Done